Libraries/DirectMLX.h

```diff
@@ -3470,9 +3475,48 @@ namespace dml
         return out;
     }
 
-    // 
-    // TODO: NonZeroCoordinates
-    // 
+    struct NonZeroCoordinatesOutputs
+    {
+        Expression count;
+        Expression coordinates;
+    };
+    inline NonZeroCoordinatesOutputs NonZeroCoordinates(Expression input)
+    {
+        detail::GraphBuilder* builder = input.Impl()->GetGraphBuilder();
+        TensorDesc inputTensor = input.Impl()->GetOutputDesc();
+        const auto& inputTensorSizes = inputTensor.sizes;
+        uint32_t dimensionCount = static_cast<uint32_t>(inputTensorSizes.size());
+
+        TensorDimensions outputSizes = {};
+        uint32_t effectiveRank = -1;
+        uint32_t totalElements = 1;
+        for (uint32_t i = 0; i < dimensionCount; ++i)
+        {
+            if (inputTensorSizes[i] > 1 && effectiveRank == -1)
+            {
+                effectiveRank = dimensionCount - i;
+            }
+            totalElements *= inputTensorSizes[i];
+            outputSizes.push_back(1);
+        }
+        TensorDesc outputCountTensor(DML_TENSOR_DATA_TYPE_UINT32, outputSizes, builder->GetTensorPolicy());
+        outputSizes[dimensionCount - 2] = totalElements;
+        outputSizes[dimensionCount - 1] = effectiveRank;
```

Comment:
I would put an assert here since it's a potential access violation (e.g. `assert(dimensionCount >= 2)`).